docs/source/extension/extension_dev.rst

```diff
@@ -45,6 +46,7 @@ Tutorials
 We provide a set of guides to get started writing extensions for JupyterLab:
 
 - :ref:`extension_tutorial`: A tutorial to learn how to make a simple JupyterLab extension.
+- :ref:`Making Dual-Compatible Extensions Tutorial <dual_compatible_extensions>` A tutorial for making extensions that work in both JupyterLab and Jupyter Notebook 7
```

Comment:
Why not call the document "Make an extension for Jupyter Notebook 7" then?

Reply:
It's not clear at first glance what "Dual-Compatible" means when just looking at the title.

Reply:
I'll switch to "Making Extensions Compatible with Multiple Applications Tutorial" to emphasize the goal, which is compatibility (@krassowski suggested something similar with the provider-consumer examples, maybe it's better to focus on the end goal).

Putting both "JupyterLab" and "Jupyter Notebook 7" makes the title excessively long, and adding the concept of Frontends in the title will probably add confusion more than clarify, as its meaning in this context will likely be unknown to those outside the circle of core devs.